src/error.c

```diff
@@ -32,6 +32,24 @@ void E_Messages()
 			fprintf(stderr,"ERROR: %s (%dx)\n",  EMessages[i], ERRORS[i]);
 }
 
+/* as above, but return "; "-delimited string instead of stderr output */
+char *E_Messages_str()
+{
+	int i;
+	size_t slen = 0;
+	char *s, *p;
+	for (i=1;i<NERR;i++)
+		if (ERRORS[i])
+			slen += (size_t) snprintf(NULL, 0, "%s (%dx); ", EMessages[i], ERRORS[i]);
+	s = p = (char *) malloc(slen + 1);
```

Comment:
It lacks a check for `NULL` before dereferencing `s`.

Reply:
I'm following the style of the rest of the library, which doesn't check for `NULL` returns from `malloc`. I don't think checking in this one spot and not in the rest of the library will serve much purpose, but I'm happy to add a check if @bartp5 wants.

(In practice, a `NULL` return will never happen for this kind of code on any modern system.   The maximum possible `slen` is less than 3kB.   You can't usefully operate any program if the memory is full to the point where a 3kB allocation fails.)